lib/LMSManagers/LMSCustomerManager.php

```diff
@@ -1587,7 +1593,14 @@ public function getCustomerList($params = array())
                     FROM nodes
                     WHERE ownerid > 0 AND ipaddr <> 0
                     GROUP BY ownerid
-                ) s ON (s.ownerid = c.id) '
+                ) s ON (s.ownerid = c.id)
+                 LEFT JOIN (SELECT a.customerid, a.id AS aid, a.dateto,
+                    a.suspended, a.commited, (CASE WHEN a.tariffid IS NOT NULL THEN t.type ELSE l.type END) as atype
+                    FROM assignments a
+                    LEFT JOIN tariffs t ON (t.id = a.tariffid)
+                    LEFT JOIN liabilities l ON (l.id = a.liabilityid AND a.period <> 0)
+                    WHERE a.commited = 1 AND a.datefrom <= ?NOW? AND (a.dateto > ?NOW? OR a.dateto = 0)
+                ) assingmentsiv ON (assingmentsiv.customerid = c.id) '
```

Comment:
To jest funkcja PHP, która zwraca listę klientów, a LEFT JOIN robimy z tabelą zobowiązań `assignments`. Co w syttuacji, gdy klient będze miał przypisane 10 zobowiązań? Czyż nie zwróci to 10 rekordów klienta i pokaże go 10 razy na liście klientów?

Reply:
No tak.
Puściłem inną wersję w https://github.com/chilek/lms/pull/2396